src/pymor/operators/constructions.py

```diff
@@ -352,6 +353,152 @@ def __rmatmul__(self, other):
+        assert len(left) == len(right)
+        assert (isinstance(core, np.ndarray)
+                and core.ndim == 2
+                and core.shape[0] == core.shape[1] == len(left))
+
+        self.__auto_init(locals())
+        self.source = right.space
+        self.range = left.space
+
+    @property
+    def H(self):
+        options = {
+            'inverse': self.solver_options.get('inverse_adjoint'),
+            'inverse_adjoint': self.solver_options.get('inverse'),
+        } if self.solver_options else None
+        return type(self)(self.right,
+                          self.core.T.conj(),
+                          self.left,
+                          inverted=self.inverted,
+                          solver_options=options,
+                          name=self.name + '_adjoint')
+
+    def apply(self, U, mu=None):
+        assert U in self.source
+        V = self.right.dot(U)
+        if self.inverted:
+            V = spla.solve(self.core, V)
+        else:
+            V = self.core @ V
+        return self.left.lincomb(V.T)
+
+    def apply_adjoint(self, V, mu=None):
+        assert V in self.range
+        U = self.left.dot(V)
+        if self.inverted:
+            U = spla.solve(self.core.T.conj(), U)
+        else:
+            U = self.core.T.conj() @ U
+        return self.right.lincomb(U.T)
+
+
+class LowRankUpdatedOperator(LincombOperator):
+    r"""|Operator| plus :class:`LowRankOperator`.
+
+    Represents a linear combination of an |Operator| and
+    :class:`LowRankOperator`. Uses the Sherman-Morrison-Woodbury formula
+    in `apply_inverse` and `apply_inverse_adjoint`:
+
+    .. math::
+        \left(\alpha A + \beta L C R^H \right)^{-1}
+        & = \alpha^{-1} A^{-1}
+            - \alpha^{-1} \beta A^{-1} L C
+            \left(\alpha C + \beta C R^H A^{-1} L C \right)^{-1}
+            C R^H A^{-1}, \\
+        \left(\alpha A + \beta L C^{-1} R^H \right)^{-1}
+        & = \alpha^{-1} A^{-1}
+            - \alpha^{-1} \beta A^{-1} L
+            \left(\alpha C + \beta R^H A^{-1} L \right)^{-1}
+            R^H A^{-1}.
+
+    Parameters
+    ----------
+    operator
+        |Operator|.
+    lr_operator
+        :class:`LowRankOperator`.
+    coeff
+        A linear coefficient for `operator`. Can either be a fixed
+        number or a |ParameterFunctional|.
+    lr_coeff
+        A linear coefficient for `lr_operator`. Can either be a fixed
+        number or a |ParameterFunctional|.
+    solver_options
+        The |solver_options| for the operator.
+    name
+        Name of the operator.
+    """
+
+    def __init__(self, operator, lr_operator, coeff, lr_coeff,
+                 solver_options=None, name=None):
+        assert isinstance(lr_operator, LowRankOperator)
+        super().__init__([operator, lr_operator], [coeff, lr_coeff],
+                         solver_options=solver_options, name=name)
+        self.__auto_init(locals())
+
+    def apply_inverse(self, V, mu=None, least_squares=False):
```

Comment:
`least_squares` is ignored. At least raise an error or call `super().apply_inverse`